Mathlib/Analysis/Normed/Ring/Seminorm.lean

```diff
@@ -388,3 +388,52 @@ lemma MulRingNorm_nat_le_nat {R : Type*} [Ring R] (n : ℕ) (f : MulRingNorm R)
       f (n + 1) ≤ f (n) + f 1 := f.add_le' ↑n 1
       _ = f (n) + 1 := by rw [map_one]
       _ ≤ n + 1 := add_le_add_right hn 1
+
+noncomputable section
+variable {R : Type*} [Ring R]
+variable {f g : MulRingNorm ℚ}
+
+/-- A multiplicative norm composed with the absolute value on integers equals the norm itself. -/
+lemma f_of_abs_eq_f (x : ℤ) (f : MulRingNorm ℚ) : f (Int.natAbs x) = f x := by
+  obtain ⟨n,rfl|rfl⟩ := Int.eq_nat_or_neg x
+  · simp only [Int.natAbs_ofNat, Int.cast_natCast]
+  · simp only [Int.natAbs_neg, Int.natAbs_ofNat, Int.cast_neg, Int.cast_natCast, map_neg_eq_map]
+
+/-- Values of a multiplicative norm of the rationals are determined by the values on the
+integers. -/
+lemma NormRat_eq_on_Int_iff_eq_on_Nat : (∀ n : ℕ , f n = g n) ↔ (∀ n : ℤ , f n = g n) := by
+  refine' ⟨_, fun a n => a n⟩
+  intro h z
+  obtain ⟨n,rfl | rfl ⟩ := Int.eq_nat_or_neg z
+  · exact h n
+  · simp only [Int.cast_neg, Int.cast_ofNat, map_neg_eq_map]
+    exact h n
+
+/-- Values of a multiplicative norm of the rationals are determined by the values on the natural
+numbers. -/
+lemma NormRat_eq_iff_eq_on_Nat : (∀ n : ℕ , f n = g n) ↔ f = g := by
+  refine' ⟨_, fun h n => congrFun (congrArg DFunLike.coe h) ↑n⟩
+  intro h
+  ext z
+  rw [← Rat.num_div_den z]
+  simp only [map_div₀]
+  rw [h, NormRat_eq_on_Int_iff_eq_on_Nat.mp h]
+
+/-- The equivalence class of a multiplicative norm on the rationals is determined by its values on
+the natural numbers. -/
+lemma NormRat_equiv_iff_equiv_on_Nat : (∃ c : ℝ, 0 < c ∧ (∀ n : ℕ , (f n)^c = g n)) ↔
```

Comment:
Once on has Ostrowski, one knows that if there is n such that f n \neq 1 and g n \neq 1, then f and g are equivalent.

Reply:
Hi! Yes, and we use this Lemma in Ostrowski's proof. :)